### PR TITLE
Add link and personal info tests

### PR DESCRIPTION
**Pull request by radyslavkrechet**

### CODE REVIEW

File: ShopAppTests/App/Account/Controllers/LinkViewControllerSpec.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            it("need to change title of email label") {
+                expect(emailLabel.text) == "user@mail.com"
+            }
+            
+            it("need to change value of view model") {
+                expect(viewModelMock.emailText.value) == "user@mail.com"
+            }
+        }
+        
+        describe("when resend button pressed") {
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/LinkViewControllerSpec.swift]: Add pressing button code

File: ShopAppTests/App/Account/Controllers/LinkViewControllerSpec.swift [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                expect(resendUnderlineView.isHidden) == true
+            }
+            
+            it("needs to show resend underline view on normal state") {
+                resendButton.isHighlighted = false
+                
+                expect(resendUnderlineView.isHidden) == false
+            }
+            
+            it("needs to resend password") {
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/LinkViewControllerSpec.swift]: Split this test case to validate success and failure cases

File: ShopAppTests/App/Account/Controllers/PersonalInfoViewControllerSpec.swift [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                    expect(emailTextFieldView.errorMessage).toEventually(equal("Error.InvalidEmail".localizable))
+                }
+            }
+            
+            context("if it have valid email") {
+                it("needs to save changes, show success toast and disable save changes button") {
+                    viewModelMock.makeValidEmailText()
+                    saveChangesButton.sendActions(for: .touchUpInside)
+                    
+                    expect(ToastCenter.default.currentToast).toEventuallyNot(beNil())
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/PersonalInfoViewControllerSpec.swift]: Can we validate also message of toast?

File: ShopAppTests/App/Account/Controllers/PersonalInfoViewControllerSpec.swift [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                    expect(saveChangesButton.isEnabled) == false
+                }
+            }
+        }
+        
+        describe("when change password button pressed") {
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/PersonalInfoViewControllerSpec.swift]: Add pressing button code

File: ShopAppTests/App/Account/ViewModels/Mocks/PersonalInfoViewModelMock.swift [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+//
+//  Created by Radyslav Krechet on 2/27/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+@testable import ShopApp
+
+class PersonalInfoViewModelMock: PersonalInfoViewModel {
+    func makeSaveChangesButtonEnabled() {
+        emailText.value = "u"
```
Comment (plzen) [inline ShopAppTests/App/Account/ViewModels/Mocks/PersonalInfoViewModelMock.swift]: Why you set emailText value when you can just emit event for saveChangesButtonEnabled observable?

File: ShopApp/App/Constants/ControllerIdentifiers.swift [hunk 6]
```diff
@@ -35,4 +35,6 @@ struct ControllerIdentifiers {
     static let orderList = "OrderListControllerIdentifier"
     static let orderDetails = "OrderDetailsControllerIdentifier"
     static let forgotPassword = "ForgotPasswordControllerIdentifier"
+    static let link = "LinkControllerIdentifier"
+    static let personalInfo = "PersonalInfoControllerIndetifier"
```
Comment (AntonoffEvgeniy) [inline ShopApp/App/Constants/ControllerIdentifiers.swift]: Change to "PersonalInfoControllerIdentifier"

File: ShopAppTests/App/Account/Controllers/LinkViewControllerSpec.swift [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+                expect(descriptionLabel.text) == "Label.ForgotPassword.LinkDescription".localizable
+                expect(additionalTitleLabel.text) == "Label.ForgotPassword.LinkAdditionalTitle".localizable
+                expect(additionalDescriptionLabel.text) == "Label.ForgotPassword.LinkAdditionalDescription".localizable
+            }
+            
+            it("should have resend button with correct title") {
+                expect(resendButton.title(for: .normal)) == "Button.Resend".localizable.uppercased()
+            }
+            
+            it("should have email label with correct title") {
```
Comment (AntonoffEvgeniy) [inline ShopAppTests/App/Account/Controllers/LinkViewControllerSpec.swift]: You have one separated test for emailLabel and another test for other labels

File: ShopAppTests/App/Account/Controllers/PersonalInfoViewControllerSpec.swift [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+            it("needs to hide change password underline view on highlighte state") {
```
Comment (AntonoffEvgeniy) [inline ShopAppTests/App/Account/Controllers/PersonalInfoViewControllerSpec.swift]: Error in "highlighte" word

File: ShopAppTests/App/Account/ViewModels/PersonalInfoViewModelSpec.swift [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+                    customer.email = "user@mail.com"
+                    viewModel.customer.value = customer
+                }
+                
+                context("and can change email") {
+                    beforeEach {
+                        viewModel.canChangeEmail = true
+                    }
+                    
+                    context("if it have changed eprty email") {
```
Comment (AntonoffEvgeniy) [inline ShopAppTests/App/Account/ViewModels/PersonalInfoViewModelSpec.swift]: "empty" instead of "eprty"

**Pull request merged**